### PR TITLE
Add "Messenger auto_setup" checker

### PR DESCRIPTION
**Pull request by M-arcus**

### CODE REVIEW

**Comment by schneider-felix** [comment]
Will this report a false positive for rabbitMQ?

**Comment by M-arcus** [comment]
@schneider-felix 
This check only reports for doctrine messenger transports
https://github.com/FriendsOfShopware/FroshTools/pull/300/files#diff-a8a941862fa111973d89c9f1f68dfeb9c60ebfe4e4693800650245e67d9f8180R44-R46

**Comment by schneider-felix** [comment]
Alright, got it. Can we add the same for rabbitMQ/Redis?

**Comment by M-arcus** [comment]
`RabbitMQ` and `Redis` messenger are not included by default, and so the classes for the transport are not loaded. How should this be implemented?

**Comment by shyim** [comment]
I would just check if the query string contains that `auto_setup` parameter and not look so much into detail

**Comment by schneider-felix** [comment]
> I would just check if the query string contains that `auto_setup` parameter and not look so much into detail

I agree. This is probably enough for 99% of setups.

**Comment by M-arcus** [comment]
Changed it to a check that looks for auto_setup bool in the query part of the dsn

File: src/Components/Health/Checker/PerformanceChecker/MessengerAutoSetupChecker.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        $query = [];
+        if (isset($params['query'])) {
+            \parse_str($params['query'], $query);
+        }
+
+        $query += ['auto_setup' => true];
+
+        return filter_var($query['auto_setup'], \FILTER_VALIDATE_BOOL);
+    }
+}
```
Comment (schneider-felix) [inline src/Components/Health/Checker/PerformanceChecker/MessengerAutoSetupChecker.php:54]:
```suggestion
    private function isAutoSetupEnabled(string $messageTransportDsn): bool
    {
        $queryParams = \parse_url($messageTransportDsn, \PHP_URL_QUERY);
        
        // Messenger DSN is invalid. Therefore we can't really check
        if ($params === false) {
            return false;
        }
        
        if ($params === null){
        	return true;
        }

        $query = [];
        \parse_str($queryParams, $query);

        $query += ['auto_setup' => true];

        return filter_var($query['auto_setup'], \FILTER_VALIDATE_BOOL);
    }
}
```

**Review by schneider-felix — COMMENTED** [review]
You could simplify this a bit by just extracting the query part. But other than that. LGTM

File: src/Components/Health/Checker/PerformanceChecker/MessengerAutoSetupChecker.php (hunk 1 above)
Comment (M-arcus) [reply src/Components/Health/Checker/PerformanceChecker/MessengerAutoSetupChecker.php:54]: The code was taken from the `\Symfony\Component\Messenger\Bridge\Doctrine\Transport\Connection::buildConfiguration` function

**Comment by M-arcus** [comment]
@frosh-automation fix-cs

**Comment by schneider-felix** [comment]
@M-arcus thank you very much for your contribution.

**Pull request merged**